packages/platform-server/test/incremental_hydration_spec.ts

```diff
@@ -1235,9 +1221,94 @@ describe('platform-server partial hydration integration', () => {
       expect(appHostNode.outerHTML).toContain('<span id="test">end</span>');
     });
 
+    it('when', async () => {
+      @Component({
+        selector: 'app',
+        template: `
+          <main (click)="fnA()">
+            @defer (on immediate; hydrate when iSaySo()) {
+              <article>
+                defer block rendered!
+                <span id="test" (click)="fnB()">{{value()}}</span>
+              </article>
+            } @placeholder {
+              <span>Outer block placeholder</span>
+            }
+            <button id="hydrate-me" (click)="triggerHydration()">Click Here</button>
+          </main>
+        `,
+      })
+      class SimpleComponent {
+        value = signal('start');
+        iSaySo = signal(false);
+        fnA() {}
+        triggerHydration() {
+          this.iSaySo.set(true);
+        }
+        fnB() {
+          this.value.set('end');
+        }
+        registry = inject(DEHYDRATED_BLOCK_REGISTRY);
+      }
+
+      const appId = 'custom-app-id';
+      const providers = [{provide: APP_ID, useValue: appId}];
+      const hydrationFeatures = () => [withIncrementalHydration()];
+
+      const html = await ssr(SimpleComponent, {envProviders: providers, hydrationFeatures});
+      const ssrContents = getAppContents(html);
+
+      // <main> uses "eager" `custom-app-id` namespace.
+      expect(ssrContents).toContain('<main jsaction="click:;');
+      // <div>s inside a defer block have `d0` as a namespace.
+      expect(ssrContents).toContain('<article>');
+      // Outer defer block is rendered.
+      expect(ssrContents).toContain('defer block rendered');
+
+      // Internal cleanup before we do server->client transition in this test.
+      resetTViewsFor(SimpleComponent);
+
+      ////////////////////////////////
+      const doc = getDocument();
+      const appRef = await prepareEnvironmentAndHydrate(doc, html, SimpleComponent, {
+        envProviders: [...providers],
+        hydrationFeatures,
+      });
+      const compRef = getComponentRef<SimpleComponent>(appRef);
+      const registry = compRef.instance.registry;
+      spyOn(registry, 'cleanup').and.callThrough();
+      appRef.tick();
+      await appRef.whenStable();
+
+      const appHostNode = compRef.location.nativeElement;
+
+      const article = appHostNode.querySelector('article');
+
+      expect(article.__ngDebugHydrationInfo__).toBeUndefined();
```

Comment:
It'd be great to avoid reading `__ngDebugHydrationInfo__` directly, we can use the `readHydrationInfo` helper method for this.

We can also create a utility function in tests to verify whether a given node has been hydrated, for example:

```suggestion
      verifyElementWasHydrated(article);
```

The helper method itself may look like this:

```
function verifyElementWasHydrated(element: HTMLElement) {
  if (readHydrationInfo(element)?.status !== HydrationStatus.Hydrated) {
    fail('Hydration error: the node is *not* hydrated: ' + el.outerHTML);
  }
}
```

Reply:
There's another spot just a few lines above where I was also reading this. So I've created two helper functions, one that verifies hydration and the other that verifies the inverse.